exonum/src/node/mod.rs

```diff
@@ -245,28 +247,24 @@ impl<'de> de::Deserialize<'de> for AllowOrigin {
     }
 }
 
-#[test]
-fn allow_origin_serde() {
-    fn check(text: &str, allow_origin: AllowOrigin) {
-        #[derive(Serialize, Deserialize)]
-        struct Config {
-            allow_origin: AllowOrigin,
+impl FromStr for AllowOrigin {
+    type Err = failure::Error;
+
+    fn from_str(s: &str) -> Result<Self, <Self as FromStr>::Err> {
```

Comment:
`Self::Err` is sufficient here

Reply:
This signature was generated by the IDE. 🙃